src/qibolab/backends.py

```diff
@@ -125,6 +125,13 @@ def execute_circuit(self, circuit, initial_state=None, nshots=1000):
                 "Hardware backend only supports circuits as initial states.",
             )
 
+        # This should be done in qibo side
+        # Temporary fix: overwrite the wire names
+        if not all(q in circuit.wire_names for q in self.platform.qubits):
+            circuit._wire_names = self.qubits
+
+        self.platform.wire_names = circuit.wire_names
```

Comment:
I guess at this point this is just unused, isn't it?

Reply:
[1b3f5f8](https://github.com/qiboteam/qibolab/pull/1079/commits/1b3f5f8b2ebf39494d75ace88bd402d4837ead9b)